### PR TITLE
PP-11511 surface `wallet_type` in charge responses

### PR DESCRIPTION
**Pull request by nlsteers**
## WHAT YOU DID

- renamed the `CardDetails` class to `CardDetailsFromResponse` to better distinguish between deserialising and serialising DTOs
- added new class `CardDetails` that surfaces `wallet_type` information when serialising reponses
- added new enum `Wallet`
- added new util for deserialising `wallet_type` from Ledger/Connector responses to Wallet enum
- added integration and pact tests for new scenarios

### CODE REVIEW

File: src/main/java/uk/gov/pay/api/model/CardDetails.java [hunk 1]
```diff
@@ -36,6 +38,9 @@ public class CardDetails {
     @JsonProperty("card_type")
     private final String cardType;
 
+    @JsonProperty("wallet_type")
+    private String walletType;
```
Comment (stephencdaly) [inline src/main/java/uk/gov/pay/api/model/CardDetails.java:42]: The only thing I'm slightly unsure about is that we use this class for both deserialisation of the response from connector/ledger and serialisation of the public API response. I wonder whether this is slightly confusing as someone might expect this field to be on the connector/ledger response.

I wonder whether it's clearer if we create a separate class to represent the public API response, and have a constructor method that accepts a CardDetails and the wallet type.

File: openapi/publicapi_spec.json [hunk 2]
```diff
@@ -1549,6 +1549,12 @@
             "type" : "string",
             "example" : "1234",
             "readOnly" : true
+          },
+          "wallet_type" : {
+            "type" : "string",
+            "description" : "The digital wallet type that the user paid with",
+            "enum" : [ "APPLE_PAY", "GOOGLE_PAY" ],
```
Comment (stephencdaly) [inline openapi/publicapi_spec.json:1556]: Not sure how this `enum` field appears

File: src/main/java/uk/gov/pay/api/model/CardDetails.java (hunk 1 above)
Comment (nlsteers) [reply src/main/java/uk/gov/pay/api/model/CardDetails.java:33]: I think it's confusing too, I've added a new class `CardDetailsFromResponse` to better distinguish between which DTOs are used for deserialisation/serialisation.

File: openapi/publicapi_spec.json (hunk 2 above)
Comment (nlsteers) [reply openapi/publicapi_spec.json]: have updated the schema deff in the `CardDetails` class with allowable values, still shows up as an enum though 🤷🏻‍♂️

Comment (stephencdaly) [reply openapi/publicapi_spec.json]: It's correct now, which is what matters. Before it wasn't correct, but I couldn't see how it was being generated at all

**Review by stephencdaly — DISMISSED** [review]
Looks good!

**Review by stephencdaly — APPROVED** [review]

**Pull request merged**